### PR TITLE
Adds the Ancient Cloning Lab ruin

### PR DESCRIPTION
**Pull request by XDTM**
Adds the Ancient Cloning Lab, an experimental lab with an earlier version of the current cloning technology. The lab was snapped off of a larger ship due to excessive acceleration, and has been drifting in space since then.

Features:
- The cloner itself: being an early version, many safety checks are missing. It will clone bodies even if they are alive or braindead; and instead of putting the same soul in the body it polls ghosts to control it. (Who says that the cloned person has to have the same conscience, after all?)
- Deep lore, that justifies the existence of the ruin

Also, while i was working on cloning code, i got carried away and fixed several snowflakey solutions.
Therefore:
- Cloning stasis is now handled by traits instead of chemicals.
- Heart attack immunity is now a trait. Code that checked specifically for corazone (ew) now checks for the trait instead.
- Ported the NOCRITDAMAGE trait from species to generic.
- Added a var for chems that process themselves out even without a liver. Currently limited to corazone, brine, and acids.

clonelab

### CODE REVIEW

**Comment by optimumtact** [comment]
`Cloning "brine" is now a single chem instead of a hodgepodge of chems and toxins.`
this literally removes depth from the game for no reason

**Comment by PKPenguin321** [comment]
This is seriously cool. I wanted to make emagging the cloner do something like this with SNPCs but they're super broken

**Comment by DaedalusGame** [comment]
>brine chems cannot be modified in any way ingame
>only 1u, well below the OD of any of these chems is added at a time
>this removes depth from the game

what did he mean by this

**Comment by XDTM** [comment]
@optimumtact i'd agree if those were medically used chems, but stuff like mute toxin and mime's bane don't make any sense. And the solution inside the pod is likely something more complex that bicaridine and salbutamol.

**Comment by Xhuis** [comment]
I want to recreate the Horatio from Endless Space as a changeling.

**Comment by Anonmare** [comment]
Can't wait to bring this back to the station and overrun the place with clones of the clown

**Comment by AnturK** [comment]
I mean the idea is fine but the ruin is just ugly.

**Comment by XDTM** [comment]
yeah, i kinda suck at mapping

**Comment by 81Denton** [comment]
>this literally removes depth from the game for no reason

What did he mean by this

File: code/game/machinery/exp_cloner.dm [hunk 1]
```diff
@@ -0,0 +1,293 @@
+//Experimental cloner; clones a body regardless of the owner's status, letting a ghost control it instead
+/obj/machinery/clonepod/experimental
+	name = "experimental cloning pod"
+	desc = "An ancient cloning pod, built when cloning technology wasn't perfected yet."
```
Comment (RandomMarine) [inline code/game/machinery/exp_cloner.dm]: Wasn't the station's cloning technology considered experimental back in the day?

File: _maps/RandomRuins/SpaceRuins/cloning_facility.dmm [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+"o" = (
+/obj/machinery/vending/snack/teal,
+/turf/open/floor/plasteel/airless/floorgrime,
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"p" = (
+/turf/open/floor/plasteel/airless/floorgrime,
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"q" = (
+/obj/structure/sign/directions/science{
+	icon_state = "direction_sci";
```
Comment (ShizCalev) [inline _maps/RandomRuins/SpaceRuins/cloning_facility.dmm]: dirty var

File: _maps/RandomRuins/SpaceRuins/cloning_facility.dmm [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"v" = (
+/turf/open/floor/plasteel/whiteblue/corner{
+	icon_state = "whitebluecorner";
+	dir = 4
+	},
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"w" = (
+/obj/structure/chair/office{
+	icon_state = "chair";
```
Comment (ShizCalev) [inline _maps/RandomRuins/SpaceRuins/cloning_facility.dmm]: dirty var

File: _maps/RandomRuins/SpaceRuins/cloning_facility.dmm [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+	dir = 8
+	},
+/obj/effect/turf_decal/stripes/line{
+	dir = 8
+	},
+/turf/open/floor/plasteel/white,
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"v" = (
+/turf/open/floor/plasteel/whiteblue/corner{
+	icon_state = "whitebluecorner";
```
Comment (ShizCalev) [inline _maps/RandomRuins/SpaceRuins/cloning_facility.dmm]: dirty var

File: _maps/RandomRuins/SpaceRuins/cloning_facility.dmm [hunk 5]
```diff
@@ -0,0 +1,511 @@
...
+	dir = 4
+	},
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"w" = (
+/obj/structure/chair/office{
+	icon_state = "chair";
+	dir = 1
+	},
+/turf/open/floor/plasteel/whiteblue/side{
+	icon_state = "whiteblue";
```
Comment (ShizCalev) [inline _maps/RandomRuins/SpaceRuins/cloning_facility.dmm]: dirty var

File: _maps/RandomRuins/SpaceRuins/cloning_facility.dmm [hunk 6]
```diff
@@ -0,0 +1,511 @@
...
+	dir = 1
+	},
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"x" = (
+/turf/open/floor/plasteel/whiteblue/side{
+	icon_state = "whiteblue";
```
Comment (ShizCalev) [inline _maps/RandomRuins/SpaceRuins/cloning_facility.dmm]: dirty var

File: _maps/RandomRuins/SpaceRuins/cloning_facility.dmm [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+	dir = 1
+	},
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"y" = (
+/obj/structure/chair/office{
+	icon_state = "chair";
```
Comment (ShizCalev) [inline _maps/RandomRuins/SpaceRuins/cloning_facility.dmm]: dirty var

File: _maps/RandomRuins/SpaceRuins/cloning_facility.dmm [hunk 8]
```diff
@@ -0,0 +1,511 @@
...
+	dir = 1
+	},
+/turf/open/floor/plasteel/whiteblue,
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"z" = (
+/obj/machinery/light{
+	dir = 4
+	},
+/turf/open/floor/plasteel/whiteblue/corner{
+	icon_state = "whitebluecorner";
```
Comment (ShizCalev) [inline _maps/RandomRuins/SpaceRuins/cloning_facility.dmm]: dirty var

File: _maps/RandomRuins/SpaceRuins/cloning_facility.dmm [hunk 9]
```diff
@@ -0,0 +1,511 @@
...
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"B" = (
+/turf/open/floor/plasteel/airless,
+/area/ruin/space/has_grav/powered/ancient_shuttle)
+"C" = (
+/turf/open/floor/plating/airless,
+/area/space/nearstation)
+"D" = (
+/obj/structure/chair/comfy{
+	icon_state = "comfychair";
```
Comment (ShizCalev) [inline _maps/RandomRuins/SpaceRuins/cloning_facility.dmm]: dirty var

File: code/game/machinery/exp_cloner.dm (hunk 1 above)
Comment (PKPenguin321) [reply code/game/machinery/exp_cloner.dm]: That's a good point yeah, maybe change this to something like "that predates even the experimental cloner that comes equipped with some nanotrasen stations."

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by ShizCalev** [comment]
This PR is a major canidate for atomization. 

Adding a map and reworking how cloning works along with chem systems are definitely things that shouldn't be mixed into the same PR.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 10]
```diff
@@ -1812,4 +1812,34 @@
...
+		var/mob/living/L = M
+		L.remove_trait(TRAIT_STABLEHEART, id)
+		L.remove_trait(TRAIT_EMOTEMUTE, id)
+		L.remove_trait(TRAIT_MUTE, id)
+	..()
+
+/datum/reagent/stasis_brine/on_mob_life(mob/living/M)
+	M.adjustBruteLoss(-3*REM, 0)
+	M.adjustOxyLoss(-6*REM, 0)
+	M.losebreath = 0
```
Comment (coiax) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: I mean, if you REALLY want to go all out, have some trait that disables "low health in crit" damage.

Comment (XDTM) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: oh yea i should've thought of that
actually why have a chem at all if i can just use traits all around

**Comment by Kromgar** [comment]
@optimumtact Maybe if the cloning brine had to be refilled there would be depth

**Comment by AlexTheSysop42** [comment]
What happens if you clone an antag with this? Do they still have antag status? Do cloned lings keep ling abilities?

**Comment by XDTM** [comment]
No to both, antag and changelings are mind based, and this makes a new mind.

**Comment by coiax** [comment]
I mean, I always had "limited amount of cloning juice" (whether that's biomass or brine) as an idea, making cloning more dependent on functioning logistics with other departments.

**Comment by XDTM** [comment]
Limitations on cloning would either foster other revival methods (defibbing, strange reagent, podcloning) or just make doctors give up on reviving. I'm not sure how to encourage the former.

**Comment by AlexTheSysop42** [comment]
It might be interesting (and make a bit more sense, realism-wise) if cloning a ling created a nonantag (because flooding the station with antags is probably bad) with ling abilities. It would probably get annoying fast though when there are 20 lings running around.

**Comment by Armhulen** [comment]
I'm ready to replace the station's cloning with this!

**Comment by Armhulen** [comment]
also i don't wanna be a dick but this has to be at least the billionth space ruin which is "arrrch we were a big station hunky dorey but then the evils attacked and now you're only seeing a tiny bit of what it was WOAH"

**Comment by DaedalusGame** [comment]
yeah, ok, armhulen, i guess you could call that an event

**Comment by Armhulen** [comment]
NO NO I'M NOT CRAZY. like look. we have a derelict station, a derelict station but with people, a derelict SCP station, a derelict mining station, a derelict lizard gassing station, and now we have a derelict cloning station

**Comment by DaedalusGame** [comment]
it's all part of the same station and all of those parts should get shuttle ports so you can put them back together

**Comment by AlexTheSysop42** [comment]
wait what

I only just now realized, alpha/beta station (from the people derelict) just _dissapeared_ from the left side and was the med/atmos/engi section. KSS13 (Russian derelict; has drones) has atmos, med and engi. It isnt broken off on a particular side though, but it doesn't have most non-med/engi departments that it should.

**Comment by PKPenguin321** [comment]
well what else would you find just floating around space @Armhulen

**Comment by Armhulen** [comment]
I dunno maybe I'm the derelict one

I still love this pr

**Comment by XDTM** [comment]
The lore for this is that it was a part of an experimental shuttle testing a new FTL technology; NT cheaped out on the controls as usual and the exterior wings, such as the cloning lab, broke off from the acceleration.
Not part of a huge golden-age station or something.

**Comment by Jalleo** [comment]
You know you could remap the ancient station a bit and add this onto that ruin.

**Comment by XDTM** [comment]
Consider that it would end up giving them infinite bodies to populate the station with. Probably best left as a rare gimmick.

**Comment by XDTM** [comment]
Cloning now uses traits instead of brine to keep the occupant in stasis. Ported the NOCRITDAMAGE trait from species to generic traits.

**Pull request merged**